TrafficSimulator/Assets/Statistics/RoadRegistrator.cs

```diff
@@ -0,0 +1,28 @@
+using System.Collections;
+using System.Collections.Generic;
+
+using RoadGenerator;
+using UnityEngine;
+using VehicleBrain;
+
+public class RoadRegistrator : MonoBehaviour
+{
+    private AutoDrive _autoDrive;
+    private Road _currentRoad;
+    void Awake()
+    {
+        _autoDrive = GetComponent<AutoDrive>();
```

Comment:
Should probably add some error handling here with a descriptive error if this script is placed on a GameObject without an AutoDrive component